### PR TITLE
Explicit new

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

**Comment by stefanpenner** [comment]
@rwjblue r?

File: key.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+var EMPTY_ARRAY = [];
+
+module.exports = Key;
```
Comment (rwjblue) [inline key.js]: I know this is silly, but can you put the export at the end of the file?

**Comment by rwjblue** [comment]
Other than one smallish, nit-pick looks good to me.

**Comment by stefanpenner** [comment]
> Other than one smallish, nit-pick looks good to me.

updated

**Comment by timmfin** [comment]
Could a new module be published to npm for this (`v0.5.6`)? I just finished going on a wild goose chase to find out that my custom broccoli plugin that was using `v0.5.5` was failing due since the actual `inputTree` was getting [sent to CoreObject](https://github.com/ember-cli/broccoli-caching-writer/pull/31/files#diff-168726dbe96b3ce427e7fedce31bb0bcL288) (and getting [re-assigned to `this`](https://github.com/ember-cli/core-object/blob/9138b1f55da74386b9547110ee5b6f52afcdfa65/core-object.js#L16-L22)).

**Comment by timmfin** [comment]
Hmm, nevermind... it is still in issue in master.

Here is what I was doing in my custom plugin's constructor:

```
CachingWriter.call(this, inputTree, pickKeysFrom(options, this.optionKeysForCachingWriter));
```

And since that is still just passed through to CoreObject, it is calling `CoreObject.prototype.init` with my singular `inputTree` instead of my options. Maybe most others are passing an array of inputTrees instead of a single inputTree object, so that things don't break as badly.

**Comment by timmfin** [comment]
And an even better fix for me was to only call `CachingWriter.prototype.init` directly:

```coffee
# In my constructor...
CachingWriter.prototype.init.call this, [inputTree], pickKeysFrom(options, @optionKeysForCachingWriter)
```

However, that directly contradicts the [instructions](https://github.com/ember-cli/broccoli-caching-writer#switching-from-broccoli-writer) that say to call `CachingWriter.apply(this, arguments)`.

**Pull request merged**